cmd/app/factory.go

```diff
@@ -212,7 +212,11 @@ func buildClient(ctx context.Context, accessToken string, withClientThrottling b
 
 	var client *github.Client
 	if host == "https://github.com" {
-		client = github.NewClient(oauthClient)
+		if accessToken != "" {
```

Comment:
Please, check if basic authentication will work for git handler if password is empty: https://github.com/gardener/docforge/blob/4044eb951aeb42bd6e1d072a2ea9838dd6ea991b/pkg/resourcehandlers/git/git_resource_handler.go#L107

Reply:
It seems it work just fine even with an empty password. I double checked the GitClone for that